hydromt_sfincs/workflows/merge.py

```diff
@@ -209,11 +227,9 @@ def merge_dataarrays(
         da1 = da1.raster.mask_nodata()
     ## reproject da2 and reset nodata value to match da1 nodata
     try:
-        da2 = (
-            da2.raster.reproject_like(da1, method=reproj_method)
-            .raster.mask_nodata()
-            .load()
-        )
+        da2 = da2.load()
+        da2 = da2.raster.reproject_like(da1, method=reproj_method)
+        da2 = da2.raster.mask_nodata()
```

Comment:
Before this was reprojecting first and then loading. I would only load after reprojecting to make sure you don't load the entire grid. I noticed you added a clip before calling this function, but in case it's called elsewhere this may cause memory issues.